src/main/java/org/folio/linked/data/service/impl/ResourceServiceImpl.java

```diff
@@ -257,4 +259,26 @@ private void breakEdgesAndDelete(Resource resource) {
     breakCircularEdges(resource);
     resourceRepo.delete(resource);
   }
+
+  private void refreshWork(Resource resource) {
+    if (resource.isOfType(INSTANCE)) {
+      extractWorkFromInstance(resource)
+        .ifPresent(work -> {
+          edgeRepo.findByIdTargetHash(work.getId())
+            .forEach(work::addIncomingEdge);
+          addOutgoingEdges(work, 1);
+        });
+    }
+  }
+
+  private void addOutgoingEdges(Resource resource, int edgesDepth) {
```

Comment:
However, with edgesDepth set to 1, the function runs only once, eliminating the need for a recursive chain, doesn't it?